### PR TITLE
:bug: Disable default apiserver timeout (60s) for Syncer tunnel connection

### PR DESCRIPTION
**Pull request by jmprusi**
## Summary

This PR adds the tunnel subresource (used by the syncer reverse tunnel) to the `LongRunningFunc` filter function. That is done to avoid the default timeout of 60s when accessing the tunnel subresource; this will fix the current issue where the Syncer tunnel is disconnected every minute, improving the end-user experience when accessing pods logs/exec... 

## Related issue(s)

Fixes https://github.com/kcp-dev/kcp/issues/2886

### CODE REVIEW

File: pkg/server/config.go [hunk 1]
```diff
@@ -360,6 +361,13 @@ func NewConfig(opts kcpserveroptions.CompletedOptions) (*Config, error) {
 	// Make sure to set our RequestInfoResolver that is capable of populating a RequestInfo even for /services/... URLs.
 	c.GenericConfig.RequestInfoResolver = requestinfo.NewKCPRequestInfoResolver()
 
+	if kcpfeatures.DefaultFeatureGate.Enabled(kcpfeatures.SyncerTunnel) {
+		c.GenericConfig.LongRunningFunc = genericfilters.BasicLongRunningRequestCheck(
```
[the remark this answers is not in the document]
Comment (jmprusi) [reply pkg/server/config.go]: done, PTAL!

File: pkg/server/config.go [hunk 2]
```diff
@@ -360,6 +362,13 @@ func NewConfig(opts kcpserveroptions.CompletedOptions) (*Config, error) {
 	// Make sure to set our RequestInfoResolver that is capable of populating a RequestInfo even for /services/... URLs.
 	c.GenericConfig.RequestInfoResolver = requestinfo.NewKCPRequestInfoResolver()
 
+	if kcpfeatures.DefaultFeatureGate.Enabled(kcpfeatures.SyncerTunnel) {
+		kubeBasicLongRunningRequestCheck := c.GenericConfig.LongRunningFunc
+		c.GenericConfig.LongRunningFunc = func(r *http.Request, requestInfo *request.RequestInfo) bool {
+			return genericfilters.BasicLongRunningRequestCheck(sets.NewString(""), sets.NewString("tunnel"))(r, requestInfo) || kubeBasicLongRunningRequestCheck(r, requestInfo)
```
[the remark this answers is not in the document]
Comment (jmprusi) [reply pkg/server/config.go]: Done!

File: pkg/server/config.go [hunk 3]
```diff
@@ -360,6 +362,13 @@ func NewConfig(opts kcpserveroptions.CompletedOptions) (*Config, error) {
 	// Make sure to set our RequestInfoResolver that is capable of populating a RequestInfo even for /services/... URLs.
 	c.GenericConfig.RequestInfoResolver = requestinfo.NewKCPRequestInfoResolver()
 
+	if kcpfeatures.DefaultFeatureGate.Enabled(kcpfeatures.SyncerTunnel) {
+		kubeBasicLongRunningRequestCheck := c.GenericConfig.LongRunningFunc
+		c.GenericConfig.LongRunningFunc = func(r *http.Request, requestInfo *request.RequestInfo) bool {
+			return kubeBasicLongRunningRequestCheck(r, requestInfo) || genericfilters.BasicLongRunningRequestCheck(sets.NewString(""), sets.NewString("tunnel"))(r, requestInfo)
```
Comment (ncdc) [inline pkg/server/config.go:368]: Sorry, 1 more request: could you please declare the tunnel check function outside and call it from in here, so we aren't allocating new sets for every invocation?

Comment (jmprusi) [reply pkg/server/config.go]: done!

**Review by ncdc — APPROVED** [review]

**Comment by jmprusi** [comment]
```
Error: failed to publish images: error building "ko://github.com/kcp-dev/kcp/cmd/syncer": GET https://cgr.dev/token?scope=repository%3Achainguard%2Fstatic%3Apull&service=ghcr.io: unexpected status code 503 Service Unavailable
```

/retest

**Pull request merged**